src/qml/components/Avatar.qml

```diff
@@ -1,38 +1,71 @@
 import QtQuick 2.12
+import QtGraphicalEffects 1.0
 
 import "../theme"
 import "./CommonHelpers"
 
-TextInCircle {
+Item {
     property string nickname
+    property string avatarUrl
 
-    color: Theme.avatarBgColor // intToHexColor(hashCode(nickname))
-    diameter: Theme.avatarHeight
-    content: nickname.replace("_", "").substring(0, 2).toUpperCase()
-    pointSize: UiHelper.fixFontSz(0.4 * diameter)
-
-    // Hash any string into an integer value
-    // Then we'll use the int and convert to hex.
-    function hashCode(str) {
-        var hash = 0;
-        for (var i = 0; i < str.length; i++) {
-            hash = str.charCodeAt(i) + ((hash << 5) - hash);
+    width: Theme.avatarHeight
+    height: width
+
+    TextInCircle {
+        color: Theme.avatarBgColor // intToHexColor(hashCode(nickname))
+        anchors.centerIn: parent
+        diameter: parent.height
+        content: nickname.replace("_", "").substring(0, 2).toUpperCase()
+        pointSize: UiHelper.fixFontSz(0.4 * diameter)
+
+        // Hash any string into an integer value
+        // Then we'll use the int and convert to hex.
+        function hashCode(str) {
+            var hash = 0;
+            for (var i = 0; i < str.length; i++) {
+                hash = str.charCodeAt(i) + ((hash << 5) - hash);
+            }
+            return hash;
+        }
+
+        // Convert an int to hexadecimal with a max length
+        // of six characters.
+        function intToHexColor(i) {
+            var hex = ((i>>24)&0xFF).toString(16) +
+                    ((i>>16)&0xFF).toString(16) +
+                    ((i>>8)&0xFF).toString(16) +
+                    (i&0xFF).toString(16);
+
+            // Sometimes the string returned will be too short so we
+            // add zeros to pad it out, which later get removed if
+            // the length is greater than six.
+            hex += '000000';
+            return "#" + hex.substring(0, 6);
         }
-        return hash;
     }
 
-    // Convert an int to hexadecimal with a max length
-    // of six characters.
-    function intToHexColor(i) {
-        var hex = ((i>>24)&0xFF).toString(16) +
-                ((i>>16)&0xFF).toString(16) +
-                ((i>>8)&0xFF).toString(16) +
-                (i&0xFF).toString(16);
-
-        // Sometimes the string returned will be too short so we
-        // add zeros to pad it out, which later get removed if
-        // the length is greater than six.
-        hex += '000000';
-        return "#" + hex.substring(0, 6);
+    Item {
+        id: imageItem
+        anchors.fill: parent
+        visible: avatarUrl !== undefined
```

Comment:
Be careful with JS comparision: https://askubuntu.com/questions/527799/how-do-you-check-if-a-property-is-undefined-in-qml
Use safer way from answer

Reply:
Thank you. I have changed my code